components/dashboard/src/admin/License.tsx

```diff
@@ -0,0 +1,215 @@
+    switch (license.plan) {
+        case "prod":
+        case "trial":
+            return professionalPlan(license.userCount || 0, license.seats, license.plan == "trial", license.validUntil);
+        case "community":
+            return communityPlan(license.userCount || 0, license.seats, license.fallbackAllowed);
+        default: {
+            return defaultMessage();
+        }
+    }
+}
+
+function licenseLevel(level: string): ReactElement {
+    return <p className="text-white dark:text-black font-bold pt-4"> {level}</p>;
+}
+
+function additionalLicenseInfo(data: string): ReactElement {
+    return <p className="dark:text-gray-500">{data}</p>;
+}
+
+function defaultMessage(): ReactElement[] {
+    const alertMessage = () => {
+        return (
+            <>
+                <div>Inactive or unknown license</div>
+                <div className="flex justify-right my-4 mx-1">
+                    <Alert fill="grey" className="h-8 w-8" />
+                </div>
+            </>
+        );
+    };
+
+    return [licenseLevel("Inactive"), additionalLicenseInfo("Free"), alertMessage()];
+}
+
+function professionalPlan(userCount: number, seats: number, trial: boolean, validUntil: string): ReactElement[] {
+    const alertMessage = (aboveLimit: boolean) => {
+        return aboveLimit ? (
+            <>
+                <div className="text-red-600">You have exceeded the usage limit.</div>
+                <div className="flex justify-right my-4 mx-1">
+                    <Alert fill="red" className="h-6 w-6" />
+                </div>
+            </>
+        ) : (
+            <>
+                <div className="text-green-600">You have an active professional license.</div>
+                <div className="flex justify-right my-4 mx-1">
+                    <Success fill="green" className="h-8 w-8" />
+                </div>
+            </>
+        );
+    };
+
+    const aboveLimit: boolean = userCount > seats;
+
+    const licenseTitle = () => {
+        if (trial) {
+            const expDate = new Date(validUntil);
+            if (typeof expDate.getTime !== "function") {
+                return additionalLicenseInfo("Trial");
+            } else {
+                return additionalLicenseInfo("Trial expires on " + expDate.toLocaleDateString());
+            }
+        } else {
+            return additionalLicenseInfo("Paid");
+        }
+    };
+
+    return [licenseLevel("Professional"), licenseTitle(), alertMessage(aboveLimit)];
+}
+
+function communityPlan(userCount: number, seats: number, fallbackAllowed: boolean): ReactElement[] {
+    const alertMessage = (aboveLimit: boolean) => {
+        if (aboveLimit) {
+            return fallbackAllowed ? (
+                <>
+                    <div>No active license. You are using community edition.</div>
+                    <div className="flex justify-right my-4 mx-1">
+                        <Success fill="grey" className="h-8 w-8" />
```

Comment:
**question**: This is minor but could we safely use the same color for the icon as the text next to the icon here as seen in the specs?

- For the gray version, we could use `gray-600` for the light theme and `gray-50` for the dark theme.
- For the green version, we could use `green-600` for the light theme and `green-400` for the dark theme.
- For the red version, we could use  `red-700` for the light theme and `red-400` for the dark theme.

Reply:
done!